src/daemon/rpmostreed-os.cxx

```diff
@@ -951,6 +951,7 @@ os_create_dnf_context_simple (RPMOSTreeOS *interface, gboolean with_sack, GCance
       && !rpmostree_context_download_metadata (
           ctx,
           static_cast<DnfContextSetupSackFlags> (DNF_CONTEXT_SETUP_SACK_FLAG_SKIP_RPMDB
+                                                 | DNF_CONTEXT_SETUP_SACK_FLAG_SKIP_FILELISTS
```

Comment:
The following is the trail of nested functions affected by setting the `DNF_CONTEXT_SETUP_SACK_FLAG_SKIP_FILELISTS` flag:

- os_create_dnf_context_simple()
    - os_handle_list_repos()
        - Doesn't take in arguments and, thus, will never take in a filename pattern as input
    - os_handle_what_provides()
        - Doesn't need filelists. Calling the command directly through dbus with a filename pattern input returns no results.
    - os_handle_get_packages()
        - Doesn't need filelists. Same reason as os_handle_what_provides()
    - os_handle_search()
        - Doesn't need filelists. Only looks at package names in repos, not filename paths.
 
 For the reason listed above, disabling filelists from being downloaded in this function should not create and issues.

Reply:
Nice work doing this analysis!  I am inclined to believe you but...here's my strawman:

Let's go ahead and aim to do something like this patch, but let's add an opt-in config (environment variable, CLI option, flag for `/etc/rpm-ostreed.conf`) that disables the filelist optimization.

My concern given all the complexity here is that it seems not unlikely that we miss a path, and if we do...particularly on the client side at least someone can set that option and get unstuck.  WDYT?

Reply:
I think adding an opt-in config is a really good idea! Missing a path or misinterpreting the effect of implementing this change was definitely a big concern for me, but I think this is a good way forward. Hopefully, as users start to play with it in a day-to-day context, they can also report bugs that we can fix.

I'll look into implementing this.

Reply:
> disables the filelist optimization
Oh just to clarify, if a user sets the flag, it would return to the existing behaviour correct? ie. by default, we are disabling filelists from being downloaded, but setting the flag would start downloading them again?

Reply:
Right.